packages/hoppscotch-app/helpers/import-export/export/openapi.ts

```diff
@@ -0,0 +1,480 @@
+  (contentType: Exclude<HoppRESTReqBody["contentType"], null>) =>
+  (body: object): OpenAPIV3.RequestBodyObject => ({
+    content: {
+      [contentType]: {
+        schema: {
+          properties: generateEntries(body),
+        },
+      },
+    },
+  })
+
+const generateOpenAPIBodyForUrlEncoded = (
+  body: string
+): OpenAPIV3.RequestBodyObject =>
+  pipe(
+    Array.from(new URLSearchParams(body)),
+    A.map(([key, value]): [string, OpenAPIV3.SchemaObject] => [
+      key,
+      {
+        default: value,
+      },
+    ]),
+    tupleToRecord,
+    (properties) => ({
+      content: {
+        "application/x-www-form-urlencoded": {
+          schema: {
+            properties,
+          },
+        },
+      },
+    })
+  )
+
+const generateOpenAPIBodyForPlainText = (
+  hoppRequestBody: HoppRESTReqBodyNonFormData
+): OpenAPIV3.RequestBodyObject => ({
+  content: {
+    "text/plain": {
+      schema: {
+        type: "string",
+        default: hoppRequestBody.body,
+      },
+    },
+  },
+})
+
+const generateOpenAPIBodyForFormData = (
+  hoppRequestBody: HoppRESTReqBodyFormData
+) =>
+  pipe(
+    hoppRequestBody,
+    (body): OpenAPIV3.RequestBodyObject => ({
+      content: {
+        "multipart/form-data": {
+          schema: {
+            properties: generateFormDataEntries(body),
+          },
+        },
+      },
+    })
+  )
+
+const isContentTypeJSON = (
+  contentType: Exclude<HoppRESTReqBody["contentType"], null>
+) => knownContentTypes[contentType] === "json"
+
+const isContentTypeUrlEncoded = (hoppRequestBody: HoppRESTReqBody) =>
+  hoppRequestBody.contentType === "application/x-www-form-urlencoded"
+
+/**
+ * generates the openapi document body for different content types
+ * 1. when the body and content type are null, we return an empty openapi body
+ * 2. when the content type is an invalid one, we return INVALID_CONTENT_TYPE error
+ * 3. when the content type is json based content types, we convert that to openapi properties recursively
+ * 4. multipart/formdata is also handled differently
+ * 5. in other valid content types, the body is handled as a string
+ */
+export const generateOpenApiRequestBody = (
+  hoppRequestBody: HoppRESTReqBody
+): E.Either<RequestBodyGenerationError, OpenAPIV3.RequestBodyObject> =>
+  pipe(
+    hoppRequestBody,
+    E.fromPredicate(isEmptyBody, () => "NON_EMPTY_BODY" as const),
+    E.map(generateOpenAPIBodyForEmptyBody),
+    E.altW(() =>
+      pipe(
+        hoppRequestBody.contentType,
+        E.fromPredicate(
+          isValidContentType,
+          () => "INVALID_CONTENT_TYPE" as const
+        ),
+        E.chainW((contentType) =>
+          pipe(
+            hoppRequestBody,
+            O.fromPredicate(isNonFormDataBody),
+            O.chain((hoppRequestBody) =>
+              pipe(
+                hoppRequestBody,
+                O.fromPredicate(() => isContentTypeJSON(contentType)),
+                O.chain(flow(({ body }) => safeParseJSON(body))),
```

Comment:
You can drop the flow here.

```suggestion
                O.chain(({ body }) => safeParseJSON(body)),
```